apps/src/aichat/views/bot/MovingBot.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {useState} from 'react';
+import Bot, {BotState} from './Bot';
+
+function pairwise(arr: [], func: Function) {
+  for (var i = 0; i < arr.length - 1; i++) {
+    func(arr[i], arr[i + 1]);
+  }
+}
+
+interface MovingBotProps {
+  currentTick: number;
+  moves: any;
```

Comment:
Looks like the move type is defined in another file and can be imported.

```suggestion
  moves: MoveBlockStep[];
```